src/Processors/Formats/Impl/JSONEachRowRowInputFormat.h

```diff
@@ -55,7 +55,13 @@ class JSONEachRowRowInputFormat : public IRowInputFormat
 
     virtual void readRowStart(MutableColumns &) {}
     virtual void skipRowStart() {}
-
+    void transformFieldNameToLowerCase(StringRef & field_name)
+    {
+        if (!field_name.data) return;
+        char * name_data = const_cast<char*>(field_name.data);
```

Comment:
I think that const_cast is a not good idea when you are trying to mutate sth that you don't own. I suggest the following: if key ignore_key_case is set to True, we will create map from the lower_case versions names to the actual names, and make more profound check while finding the column_name in our set